tests/helpers/flows/nc.ts

```diff
@@ -0,0 +1,198 @@
+  } catch (error) {
+    return {
+      success: false,
+      step: 'nc-session-setup',
+      error: error as Error
+    };
+  }
+}
+
+/**
+ * Completes a full application for the NC white label using the provided test data
+ * @param page - Playwright page instance
+ * @param data - Application data to use for the test
+ * @returns Promise with flow result
+ */
+export async function completeNcFullApplication(
+  page: Page,
+  data: ApplicationData
+): Promise<FlowResult> {
+  try {
+    // Complete location info
+    const locationResult = await completeLocationInfo(page, data.zipCode, data.county);
+    if (!locationResult.success) return locationResult;
+    
+    // Complete household size
+    const householdSizeResult = await completeHouseholdSize(page, data.householdSize);
+    if (!householdSizeResult.success) return householdSizeResult;
+    
+    // Complete primary user info
+    const primaryUserResult = await completePrimaryUserInfo(page, data.primaryUser);
+    if (!primaryUserResult.success) return primaryUserResult;
+    
+    // Complete household member info
+    const memberResult = await completeHouseholdMemberInfo(page, data.householdMember);
+    if (!memberResult.success) return memberResult;
+    
+    // Complete expenses
+    const expensesResult = await completeExpenses(page, data.expenses);
+    if (!expensesResult.success) return expensesResult;
+    
+    // Complete assets
+    const assetsResult = await completeAssets(page, data.assets);
+    if (!assetsResult.success) return assetsResult;
+    
+    // Complete public benefits (just continue)
+    const benefitsResult = await completePublicBenefits(page);
+    if (!benefitsResult.success) return benefitsResult;
+    
+    // Complete needs
+    const needsResult = await completeNeeds(page, data.needs);
+    if (!needsResult.success) return needsResult;
+    
+    // Complete referral source
+    const referralResult = await completeReferralSource(page, data.referralSource);
+    if (!referralResult.success) return referralResult;
+    
+    // Complete additional info
+    const additionalInfoResult = await completeAdditionalInfo(page);
+    if (!additionalInfoResult.success) return additionalInfoResult;
+    
+    // Navigate to results
+    const resultsResult = await navigateToResults(page);
+    if (!resultsResult.success) return resultsResult;
+    
+    return { success: true, step: 'nc-full-application' };
+  } catch (error) {
+    return {
+      success: false,
+      step: 'nc-full-application',
+      error: error as Error
+    };
+  }
+}
+
+/**
+ * Runs a complete end-to-end test for the NC white label
+ * @param page - Playwright page instance
+ * @param data - Application data to use for the test
+ * @returns Promise with flow result
+ */
+export async function runNcEndToEndTest(
+  page: Page,
+  data: ApplicationData
+): Promise<FlowResult> {
+  // Add retry mechanism for more resilience
+  const maxRetries = 2;
```

Comment:
I guess I'm thinking rather than add a retry mechanism I would opt for adding more specific waits. If we do decide to go with a retry mechanism we could move this somewhere common.